### ISSUE TITLE
[FEATURE REQUEST] Add named events

### ISSUE THREAD

**Issue by go4cas** [elysiajs/stream#1]
It would be great to have the `event` field (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event) added to the SSE stream, so that clients can subscribe to specific named events.

**Comment by go4cas** [comment]
I have discover that this already technically possible, if the `event` is added to the `data`, and prefixed with the `id`:
https://github.com/elysiajs/stream/blob/f42d186d93e960acab7a6d545be918182da2d3a4/src/index.ts#L106-L107

So, something like this will work:
```javascript
const eventData = `id: ${nanoid()}\nevent: myevent\ndata: Message to my named event\n\n`
stream.send(eventData)
```

It does, however, feel a bit hacky, so it would be great if the event name could be passed in as a parameter to the `send` method.

Added Bonus: The `EventSource` spec also caters for [retry](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#retry) - would be a nice addition.

**Comment by SaltyAom** [comment]
[a210dc4](https://github.com/elysiajs/stream/commit/a210dc4b2c0d200ceae0d0a910a5f000933097aa) published in 0.7.2
